phastSim/phastSim.py

```diff
@@ -607,14 +693,96 @@ def normalize(node, norm):
+            # if there are indels then some further calculation is require to convert the rates from 
+            # substitution rates to overall rates.
+            
+
+            def include_indels(node, scale):
+                
+                if node.isTerminal:
+                    node.insertionRate * scale
+                    node.deletionRate * scale
+                    node.rate += (node.insertionRate + node.deletionRate)
+                
+                else:
+                    node.rate = include_indels(node.belowNodes[0], scale) + include_indels(node.belowNodes[1], scale)
+
+                return node.rate
+
+            include_indels(node=self.genomeRoot, scale=scale)
+        """
+        # TODO Check this - is it right?! It's the same as the original code but seems weird. 
+        # Why are we updating self.norm to a value that is just about to get rescaled?!
+        norm = self.genomeRoot.rate / len(self.ref)
+        self.norm = norm
+
+        print("\n Total cumulative substitution rate per site before normalization: " + str(norm))
+        self.normalizeRates(self.genomeRoot, self.indels, self.codon, len(self.ref), scale)
+
+    def normalizeRates(self, rootNode, indels, codon, genome_length, scale):
+        # This is an internal function implementation that can be reused elsewhere (e.g. when creating indel inserts).
 
+        # I am assuming the branch lengths are in number of substitutions per nucleotide,
+        # even though we might be simulating a codon model.
+        norm = rootNode.rate / genome_length
+
+        # We rescale by the input normalization factor, this is the same as rescaling all
+        # the branch lengths by this rescaling factor
+        norm = norm / scale
+
+        # define a function either for codon or nucleotide mode
+        if codon:
+            # function to iteratively normalize all rates
+            def normalize(node, norm):
+                node.rate /= norm
+                if node.isTerminal:
+                    node.rates[node.allele] /= norm
+
+                else:
+                    normalize(node.belowNodes[0], norm)
+                    normalize(node.belowNodes[1], norm)
+
+        else:
+            # function to iteratively normalize all rates
+            def normalize(node, norm):
+                node.rate /= norm
+                if node.isTerminal:
+                    node.rates /= norm
+
+                else:
+                    normalize(node.belowNodes[0], norm)
+                    normalize(node.belowNodes[1], norm)
+
+        # normalize the rates
+        normalize(node=rootNode, norm=norm)
+
+        if indels:
+            # if there are indels then some further calculation is require to convert the rates from 
+            # substitution rates to overall rates.
+            
+
+            def include_indels(node, scale):
+                
+                if node.isTerminal:
+                    node.insertionRate * scale
```

Comment:
Same as before, here probably it should be *=

Reply:
Yes thank you - and here it is actually in a real function and not commented out!